File: src/benchmarks/simplebenchmarks/nsieve/nsieve.ml
let nsieve n =
  let count = ref 0 in
  let flags = Array.make n true in
  for i = 2 to n - 1 do
    if flags.(i) then (
      let j = ref (i lsl 1) in
      while !j < n do
        flags.(!j) <- false ;
        j := !j + i
      done ;
      count := !count + 1 )
    else ()
  done ;
  Printf.printf "Primes up to %d %d\n" n !count

let get_n =
  if Array.length Sys.argv > 1 then
    try int_of_string Sys.argv.(1) with _ -> 10
  else 10

let () =
  let n = get_n in
  nsieve n

File: src/benchmarks/simplebenchmarks/nsieve/nsieve.py
def main(n):
    flags = [True]*n
    count = 0
    for i in range(2, n):
        if flags[i]:
            j = i << 1
            while j < n:
                flags[j] = False
                j += i
            count += 1
    print("Primes up to " + str(n) + " " + str(count))


import sys
n = int(sys.argv[1])
main(n)